MAINTAINERS.md

```diff
@@ -1,14 +1,46 @@
+## Introduction
 
-The idea behind Stackage is that, if all packages work with the newest versions of dependencies, we avoid dependency hell. Specifically, we aim for:
+Stackage provides a series of snapshots: a set of tuples of packages and exact versions that is free of incompatibilities. It relies on two parties: Firstly repeated automated testing ensures that incompatibilities with newly released packages are detected, and secondly maintainers update their packages to keep them up to date with their dependencies.
 
-* All packages are buildable and testable from Hackage. We recommend [the Stack Travis script](http://docs.haskellstack.org/en/stable/GUIDE.html#travis-with-caching), which ensures a package is not accidentally incomplete.
-* All packages are compatible with the newest versions of all dependencies (You can find restrictive upper bounds by visiting http://packdeps.haskellers.com/feed?needle=PACKAGENAME).
-* All packages in a snapshot are compatible with the versions of libraries that ship with the GHC used in the snapshot ([more information on lenient lower bounds](https://www.fpcomplete.com/blog/2014/05/lenient-lower-bounds)).
+Membership in any stackage snapshot is not reflected in a package's content; the stackage repository contains the metadata that determines membership. The repository relies on the following social contract in the hopes of enabling a stable set of packages, optimally never dropping any packages from newer snapshots.
 
-## Adding a package
+# Agreement
+
+1. In the following, the term "package maintainer" refers to the person responsible for Stackage membership of said package. Such a person need not be the hackage maintainer of the package, although it certainly makes sense for the stackage-side maintainer to have write access should the need for updating a package arise. Please consider contacting the hackage maintainer(s) if you wish adding a package to Stackage.
+
+2. The following conditions apply to any package included in the (next) snapshot:
+
+    * It must be compatible with the latest versions of all of its dependencies;
+    * It must be compatible with the versions of libraries (aka `bootlibs`) shipping with the GHC connected to the snapshot (see the note below);
+    * It must compile and test successfully with the usual automated building infrastructure (e.g. by the hackage buildbot);
+    * (It probably should also be compatible with the latest published snapshot.)
+
+3. The package maintainer is expected to ensure that a package continues to meet the above requirements by updating/fixing a package when dependencies are updated. The recommended timeframe for such fixes are:
+
+    * **Up to one week** if restrictive version bounds are the only problem;
+    * **Between a week and a month**, depending on the work required, if the cause of the issue is a breaking change in a dependency.
```

Comment:
Previously "causing a breaking change", which sounded like "breaking change of the maintained package. This phrasing better captures the intended meaning, or do i misunderstand something?